docs/src/main/tut/FeatureOverview.md

```diff
@@ -215,6 +215,190 @@ case class PriceInfo3(ratio: Int, price: Double) // ratio should be Double
+```tut:silent
+case class ApartmentShortInfo(city: String, price: Double, bedrooms: Int)
+
+val aptTypedDs2: TypedDataset[ApartmentShortInfo] = aptTypedDs.project[ApartmentShortInfo]
+```
+
+The union of `aptTypedDs2` with `aptTypedDs` uses all the fields of the caller (`aptTypedDs2`)
+and expects the other (`aptTypedDs`) dataset to include all those fields. 
+If field names/types do not much you get a compilation error. 
+
+```tut:book
+aptTypedDs2.union(aptTypedDs).show().run
+```
+
+The other way around will not compile, since `aptTypedDs2` has only a subset of the fields. 
+
+```tut:book:fail
+aptTypedDs.union(aptTypedDs2).show().run
+```
+
+Finally, as with `project`, `union` will align fields that have same names/types,
+so fields do not have to be in the same order. 
+
+## TypedDataset functions and transformations
+
+Frameless supports many of Spark's functions and transformations. 
+However, whenever a Spark function does not exist in Frameless, 
+calling the `.dataset` will take you back to vanilla `Dataset` where
+you can apply the function you need.  
+
+These are the main imports for Frameless' aggregate and non-aggregate functions.
+
+```scala
+import frameless.functions._                // For literals
+import frameless.functions.nonAggregate._   // e.g., concat, abs
+import frameless.functions.aggregate._      // e.g., count, sum, avg 
+```
+
+### Drop/Replace/Add fields
+
+`dropTupled()` drops a single column and results in a tuple-based schema.
+
+```tut:book
+aptTypedDs2.dropTupled('price): TypedDataset[(String,Int)]
+```
+
+To drop a column and specify a new schema use `drop()`.
+
+```tut:book
+case class CityBeds(city: String, bedrooms: Int)
+val cityBeds: TypedDataset[CityBeds] = aptTypedDs2.drop[CityBeds] 
+```
+
+Often, you want to replace an existing column with a new value.
+ 
+```tut:book
+val inflation = aptTypedDs2.withColumnReplaced('price, aptTypedDs2('price) * 2)
+ 
+inflation.show(2).run()
+```
+
+Or use a literal instead.
+
+```tut:book
+import frameless.functions.lit
+aptTypedDs2.withColumnReplaced('price, lit(0.001)) 
+```
+
+Adding a column using `withColumnTupled()` results in a tupled-based schema.
+
+```tut:book
+aptTypedDs2.withColumnTupled(lit(Array("a","b","c"))).show(2).run()
+```
+
+Similarly, `withColumn()` adds a column and explicitly expects a schema for the result.
+
+```tut:book
+case class CityBedsOther(city: String, bedrooms: Int, other: List[String])
+
+cityBeds.
+   withColumn[CityBedsOther](lit(List("a","b","c"))).
+   show(1).run()
+```
+
+To conditionally change a column use the `when/otherwise` operation. 
+
+```tut:book
+import frameless.functions.nonAggregate.when
+aptTypedDs2.withColumnTupled(
+   when(aptTypedDs2('city) === "Paris", aptTypedDs2('price)).
+   when(aptTypedDs2('city) === "Lyon", lit(1.1)).
+   otherwise(lit(0.0))).show(8).run()
+```
+
+A simple way to add a column without loosing important schema information is
+to project the entire source schema into a single column. 
```

Comment:
We should maybe note that his is something new that is not part of the vanilla APIs